### PR TITLE
dropbox_uploader 0.16 (new formula)

### PR DESCRIPTION
**Pull request by iandrewt**
### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

### New Formulae Submissions:

- [x] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Have you built your formula locally prior to submission with `brew install <formula>`?

dropbox_uploader is a bash script for interacting with Dropbox.
the `test do` portion is just `true` because the script requires API keys from dropbox (obtainable by user) for first run, which is interative

### CODE REVIEW

File: Library/Formula/dropbox-uploader.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class DropboxUploader < Formula
+  desc "Bash script for interacting with Dropbox"
+  homepage "http://www.andreafabrizi.it/?dropbox_uploader"
+  url "https://github.com/andreafabrizi/Dropbox-Uploader/archive/0.16.tar.gz"
+  sha256 "738aebdb57f75e9033cb9bb319b14fe1c3b471f951180acb48631e9810ebf1c3"
+
+  depends_on "curl"
```
Comment (DomT4) [inline Library/Formula/dropbox-uploader.rb]: OS X ships `curl` by default. This shouldn't be required, unless upstream only works with specific versions of `curl`?

File: Library/Formula/dropbox-uploader.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  def install
+    bin.install "dropbox_uploader.sh"
+  end
+
+  test do
+    system "true"
```
Comment (DomT4) [inline Library/Formula/dropbox-uploader.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/dropbox-uploader.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class DropboxUploader < Formula
+  desc "Bash script for interacting with Dropbox"
+  homepage "http://www.andreafabrizi.it/?dropbox_uploader"
+  url "https://github.com/andreafabrizi/Dropbox-Uploader/archive/0.16.tar.gz"
+  sha256 "738aebdb57f75e9033cb9bb319b14fe1c3b471f951180acb48631e9810ebf1c3"
+
+  def install
+    bin.install "dropbox_uploader.sh"
```
Comment (bfontaine) [inline Library/Formula/dropbox-uploader.rb]: It’s probably better to have it named `dropbox_uploader` (without the extension). To install it under this name use:
```rb
bin.install "dropbox_uploader.sh" => "dropbox_uploader"
```

File: Library/Formula/dropbox-uploader.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  test do
+    (testpath/".dropbox_uploader").write("APPKEY=a\nAPPSECRET=b\nACCESS_LEVEL=sandbox\nOAUTH_ACCESS_TOKEN=c\nOAUTH_ACCESS_TOKEN_SECRET=d")
+    system("yes | dropbox_uploader.sh unlink")
```
Comment (bfontaine) [inline Library/Formula/dropbox-uploader.rb]: Use `pipe_output("#{bin}/dropbox_uploader.sh unlink", "y\n")`.

File: Library/Formula/dropbox-uploader.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  homepage "http://www.andreafabrizi.it/?dropbox_uploader"
+  url "https://github.com/andreafabrizi/Dropbox-Uploader/archive/0.16.tar.gz"
+  sha256 "738aebdb57f75e9033cb9bb319b14fe1c3b471f951180acb48631e9810ebf1c3"
+
+  def install
+    bin.install "dropbox_uploader.sh"
+  end
+
+  test do
+    (testpath/".dropbox_uploader").write("APPKEY=a\nAPPSECRET=b\nACCESS_LEVEL=sandbox\nOAUTH_ACCESS_TOKEN=c\nOAUTH_ACCESS_TOKEN_SECRET=d")
```
Comment (bfontaine) [inline Library/Formula/dropbox-uploader.rb]: You can split this on multiple lines with:
```rb
(testpath/"dropbox_uploader").write <<-EOS.undent
  APPKEY=a
  APPSECRET=b
  ...
EOS

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/247eb83233976fb98bcaa4829c01bba5636f51d9. Thank you for your contribution to Homebrew @iandrewt; we appreciate it! :smiley_cat:

**Pull request closed without merging**